openrouteservice/src/main/java/org/heigit/ors/matrix/algorithms/core/CoreMatrixAlgorithm.java

```diff
@@ -0,0 +1,538 @@
+     */
+    private float[][] swapResults(MatrixLocations srcData, MatrixLocations dstData, float[] times, float[] distances, float[] weights) {
+        boolean hasTimes = times != null;
+        boolean hasDistances = distances != null;
+        boolean hasWeights = weights != null;
+        float[] newTimes = new float[0];
+        float[] newDistances = new float[0];
+        float[] newWeights = new float[0];
+        if(hasTimes)
+            newTimes = new float[times.length];
+        if(hasDistances)
+            newDistances = new float[distances.length];
+        if(hasWeights)
+            newWeights = new float[weights.length];
+
+        int i = 0;
+        int srcSize = srcData.size();
+        int dstSize = dstData.size();
+        for (int dst = 0; dst < dstSize; dst++){
+            for(int src = 0; src < srcSize; src++){
+                if(hasTimes)
+                    newTimes[dst + src * dstSize] = times[i];
+                if(hasDistances)
+                    newDistances[dst + src * dstSize] = distances[i];
+                if(hasWeights)
+                    newWeights[dst + src * dstSize] = weights[i];
+                i++;
+            }
+        }
+        return new float[][]{newTimes, newDistances, newWeights};
+    }
+
+    private void extractMetrics(MatrixLocations srcData, MatrixLocations dstData, float[] times, float[] distances, float[] weights) throws Exception {
+        AveragedMultiTreeSPEntry[] destTrees = new AveragedMultiTreeSPEntry[dstData.size()];
+        for (int i = 0; i < dstData.size(); i++)
+            destTrees[i] = targetMap.get(dstData.getNodeIds()[i]);
+
+        AveragedMultiTreeSPEntry[] originalDestTrees = new AveragedMultiTreeSPEntry[dstData.size()];
+
+        int j = 0;
+        for (int i = 0; i < dstData.size(); i++) {
+            if (dstData.getNodeIds()[i] != -1) {
+                originalDestTrees[i] = destTrees[j];
+                ++j;
+            } else {
+                originalDestTrees[i] = null;
+            }
+        }
+        pathMetricsExtractor.setSwap(swap);
+        pathMetricsExtractor.calcValues(originalDestTrees, srcData, dstData, times, distances, weights);
+    }
+
+    boolean isCoreNode(int node) {
+        return chGraph.getLevel(node) >= coreNodeLevel;
+    }
+
+    boolean considerTurnRestrictions() {
+        if (!hasTurnWeighting)
+            return false;
+        return true;
+    }
+
+    /**
+     * Check whether the turnWeighting should be in the inORS mode. If one of the edges is a virtual one, we need the original edge to get the turn restriction.
+     * If the two edges are actually virtual edges on the same original edge, we want to disable inORS mode so that they are not regarded as u turn,
+     * because the same edge id left and right of a virtual node results in a u turn
+     * @param iter
+     * @param currEdge
+     * @return
+     */
+    private boolean isInORS(EdgeIteratorState iter, AveragedMultiTreeSPEntry currEdge) {
```

Comment:
While I get the general idea explained it he comment above, I fail to see how exactly the code of this method translates to the described functionality. In particular, how does the condition `EdgeIteratorStateHelper.getOriginalEdge(iter) != EdgeIterator.NO_EDGE` relate to virtual edges and why does it differ from the condition in the other method `currEdgeItem.getOriginalEdge() != EdgeIteratorStateHelper.getOriginalEdge(iter)` which I actually understand?

Reply:
This is indeed not needed. Was a remain from dev work. It actually did not do anything. Thanks.